pkg/apis/core/v1alpha1/hardware_methods.go

```diff
@@ -0,0 +1,25 @@
+package v1alpha1
+
+const (
+	// HardwareIDAnnotation is used by the controller to store the
+	// ID assigned to the hardware by Tinkerbell for migrated hardware.
+	HardwareIDAnnotation = "hardware.tinkerbell.org/id"
+)
+
+// TinkID returns the Tinkerbell ID associated with this Hardware.
+func (t *Hardware) TinkID() string {
+	annotations := t.GetAnnotations()
+	if len(annotations) == 0 {
+		return ""
+	}
+
+	return annotations[HardwareIDAnnotation]
+}
+
+// SetTinkID sets the Tinkerbell ID associated with this Hardware.
+func (t *Hardware) SetTinkID(id string) {
+	if t.GetAnnotations() == nil {
+		t.SetAnnotations(make(map[string]string))
+	}
+	t.Annotations[HardwareIDAnnotation] = id
+}
```

Comment:
Seeing how Annotations is public why not just:

```suggestion
// TinkID returns the Tinkerbell ID associated with this Hardware.
func (t *Hardware) TinkID() string {
	return t.Annotations[HardwareIDAnnotation]
}

// SetTinkID sets the Tinkerbell ID associated with this Hardware.
func (t *Hardware) SetTinkID(id string) {
	if t.Annotations == nil {
		t.Annotations = make(map[string]string)
	}
	t.Annotations[HardwareIDAnnotation] = id
}
```

Alternatively if you want to use the Get/Set methods something like in Set* methods:
```
annotations := t.GetAnnotations()
if annotations == nil {
	annotations = make(map[string]string)
}
annotations[HardwareIDAnnotations] = id
t.SetAnnotations(annotations)
```

That way code isn't mixing and matching use of both `Get`/`Set` methods *and* direct access to the underlying map. Best to pick just one way imo (I prefer direct access as in the first)

Reply:
No objection to this, just copied from the CAPT provider. I'll update

Reply:
thanks, the original way is super weird because not only is it mixing `Get`/`Set` vs direct use but also `len(map...)` vs `map == nil` checks.

Reply:
On the `TinkID()` case, the map can be `nil`, so we need to check if it exists

Reply:
Got a link that I might be able to follow to see about submitting a PR for doing it just one way?

Reply:
> On the `TinkID()` case, the map can be `nil`, so we need to check if it exists

Actually no, go does the right thing if its nil: https://go.dev/play/p/NDqUXBkSFnd

Reply:
I couldn't remember if that was the case so went and checked earlier.